modules/Bio/EnsEMBL/Compara/RunnableDB/PrepareMaster/RenameGenome.pm

```diff
@@ -0,0 +1,159 @@
+        -xml_file $ENSEMBL_CVS_ROOT_DIR/ensembl-compara/conf/citest/mlss_conf.xml \
+        -species_tree $ENSEMBL_CVS_ROOT_DIR/ensembl-compara/conf/citest/species_tree.branch_len.nw \
+        -genome_dumps_dir /hps/nobackup2/production/ensembl/jalvarez/genome_dumps/ \
+        -sketch_dir /hps/nobackup2/production/ensembl/jalvarez/species_tree/citest_sketches/
+
+=cut
+
+package Bio::EnsEMBL::Compara::RunnableDB::PrepareMaster::RenameGenome;
+
+use warnings;
+use strict;
+
+use Bio::EnsEMBL::Hive::Utils;
+use Bio::EnsEMBL::Registry;
+
+use base ('Bio::EnsEMBL::Compara::RunnableDB::BaseRunnable');
+
+
+sub fetch_input {
+    my $self = shift;
+
+    # Get all the required compara DBAdaptors
+    my $db_list = $self->param_required('prev_dbs');
+    $db_list = [$db_list] unless ref($db_list);
+    push @{$db_list}, $self->param_required('master_db');
+    my %dba_hash = $self->get_all_compara_dbas($db_list);
+    $self->param('dba_hash', \%dba_hash);
+}
+
+
+sub run {
+    my $self = shift;
+
+    my $old_name = $self->param_required('old_name');
+    my $new_name = $self->param_required('new_name');
+    my $xml_file = $self->param_required('xml_file');
+    my $species_tree = $self->param_required('species_tree');
+    my $genome_dumps_dir = $self->param_required('genome_dumps_dir');
+    my $sketch_dir = $self->param_required('sketch_dir');
+
+    my $master_db = $self->param('master_db');
+    my $dba_hash = $self->param('dba_hash');
+    my $genome_db = $dba_hash->{$master_db}->get_GenomeDBAdaptor()->fetch_by_registry_name($old_name);
+
+    # We really need a transaction to ensure we are not screwing the databases
+    $self->call_within_transaction(sub {
+        foreach my $dba (values %{$dba_hash}) {
+            # Make sure we are ensadmin for every connection
+            $self->elevate_privileges($dba->dbc);
+            $dba->dbc->do("UPDATE genome_db SET name = '$new_name' WHERE name = '$old_name' AND first_release IS NOT NULL AND last_release IS NULL");
+            $dba->dbc->do("UPDATE method_link_species_set_tag SET value = '$new_name' WHERE tag LIKE '%reference_species' AND value = '$old_name'");
+            print "Species '$old_name' renamed to '$new_name' in ", $dba->dbc->dbname, "\n" if $self->debug;
+        }
+    });
+
+    # Rename the genome in the mlss_conf.xml and species tree files
+    my $content = $self->_slurp($xml_file);
+    $content =~ s/$old_name/$new_name/;
+    $self->_spurt($xml_file, $content);
+    print "\nUpdated content of $xml_file\n" if $self->debug;
+    $content = $self->_slurp($species_tree);
+    $content =~ s/$old_name/$new_name/;
+    $self->_spurt($species_tree, $content);
+    print "Updated content of $species_tree\n" if $self->debug;
+
+    my $subdir = Bio::EnsEMBL::Hive::Utils::dir_revhash($genome_db->dbID);
+    my $dumps_path = "$genome_dumps_dir/$subdir";
+    # Rename *.fa and *.fai files
+    my @fasta_files = glob qq(${dumps_path}/${old_name}*.fa ${dumps_path}/${old_name}*.fai);
```

Comment:
`${old_name}.*` to avoid catching other species whose name start with `$old_name`